implants/lib/transport/src/config.rs

```diff
@@ -0,0 +1,153 @@
+use anyhow::{anyhow, Result};
+use std::collections::HashMap;
+
+/// Transport configuration parsed from URI query parameters
+#[derive(Debug, Clone)]
+pub struct TransportConfig {
+    /// Full URI including query parameters (required)
```

Comment:
This should be just the base URI.
Once the DSN is parsed into a TransportConfig the config should be used not the DSN.